cli/kata-check.go

```diff
@@ -134,17 +134,33 @@ func getCPUFlags(cpuinfo string) string {
 // haveKernelModule returns true if the specified module exists
 // (either loaded or available to be loaded)
 func haveKernelModule(module string) bool {
+	kmodLog := kataLog.WithField("module", module)
+
 	// First, check to see if the module is already loaded
 	path := filepath.Join(sysModuleDir, module)
 	if katautils.FileExists(path) {
 		return true
 	}
 
+	// Kernel module can be built-in, check /lib/modules/x.x.x before
+	// trying to load the module
+	cmd := exec.Command(modProbeCmd, "-R", module)
+	if output, err := cmd.CombinedOutput(); err != nil {
+		kmodLog.WithError(err).Warnf("Module can not be inserted: %s. Is it built-in?", string(output))
```

Comment:
Maybe we should actually check if the module is built-in here by taking a look at /sys/modules.

Reply:
we are already looking for it in above chuck of code, hence this code may not be required, thanks